File: OWASPTop10/Cryptographic Failures/forgot-password.py
@app.route('/forgot-password', methods=['POST'])
def forgot_password():
    username = request.args.get('username')
    token = hashlib.md5(username.encode()).hexdigest()
    allow_password_reset(username)
    send_password_reset_email(username, token)
    return 'Check your email!'

File: vulnerabilities/Cryptographic Failures/forgot-password.js
app.post('/forgot-password', (req, res) => {
    username = req.query.username
    token = crypto.createHash('md5').update(username).digest("hex")
    allow_password_reset(username)
    send_password_reset_email(username, token)
    res.send('Check your email!')
  })